src/ui.data-list/types.ts

```diff
@@ -12,15 +12,25 @@ export interface DragMoveEvent extends DragEvent {
 }
 
 export interface ListItem {
-  children?: Array<ListItem>;
+  children?: ListItem[];
   [key: string]: unknown;
 }
 
+export interface ElementObject {
+  isActive: boolean;
+  isHiddenActions: boolean;
+  isHiddenCustomActions: boolean;
+  isHiddenDelete: boolean;
+  isHiddenEdit: boolean;
+  isDisabledNesting: boolean;
+  children: ListItem[];
```

Comment:
it is possible to have other keys also. and all this keys is optional.

and ListItem and ElementObject is the same type. should be mreged.